collector/bgp.go

```diff
@@ -160,7 +160,15 @@ func (*BGPL2VPNCollector) Describe(ch chan<- *prometheus.Desc) {
 }
 
 func getBgpL2vpnEvpnSummary() ([]byte, error) {
-	return execVtyshCommand("-c", "show evpn vni json")
+	var args []string
+
+        if vtyshSudo == true {
```

Comment:
Is there a need for this conditional? I.e. can't `args = []string{fmt.Sprintf("-c show evpn vni json")}` be used as `execVtyshCommand` handles the path of `vtysh` and `sudo`.

The above comment applies to each function that calls `execVtyshCommand`

Reply:
`CommandContext ` wait for a path of a binary and next a list of args. I didn't find a way to do it otherwise when using sudo.

Reply:
What about using the below in the `execVtyshCommand` function?

```
	if vtyshSudo == true {
		a := []string{vtyshPath}
		a = append(a, args...)
		output, err = exec.CommandContext(ctx, "/usr/bin/sudo", a...).Output()
	} else {
		output, err = exec.CommandContext(ctx, vtyshPath, args...).Output()
	}
```
 That way we don't need a sudo conditional every time time `execVtyshCommand` is called. Installed we can set `args = []string{fmt.Sprintf("-c show evpn vni json")}` regardless of whether sudo is used or not.

Reply:
Let me implement it, it's fine for me

Reply:
I have spare time now, I will update the PR in following next days hopefully.

Reply:
I implemented your comments